src/tests/functional/shared_test_classes/src/subgraph/gather_weights_decompression.cpp

```diff
@@ -0,0 +1,238 @@
+    in_data.range = 1;
+    in_data.resolution = 30000;
+    auto scale_tensor = ov::test::utils::create_and_fill_tensor(output_precision, scaleshift_const_shape, in_data);
+    for (size_t i = 0; i < scale_tensor.get_size(); i++) {
+        if (output_precision == ov::element::f16)
+            scale_tensor.data<ov::float16>()[i] /= ov::float16(16.f);
+        else if (output_precision == ov::element::f32)
+            scale_tensor.data<float>()[i] /= 16.f;
+    }
+    std::shared_ptr<ov::Node> scale_const = std::make_shared<ov::op::v0::Constant>(scale_tensor);
+    if (reshape_on_decompression_constant) {
+        auto scale_reshape_const =
+            ov::op::v0::Constant::create(ov::element::i32, {scaleshift_target_shape.size()}, scaleshift_target_shape);
+        auto scale_reshape = std::make_shared<ov::op::v1::Reshape>(scale_const, scale_reshape_const, false);
+        scale_const = scale_reshape;
+    }
+    std::shared_ptr<ov::Node> last_node = std::make_shared<ov::op::v1::Multiply>(mul_parent, scale_const);
+
+    if (group_decompression) {
+        auto reshape_target_shape = std::vector<int>{static_cast<int>(data_shape[0]), -1};
+        auto target_shape_node =
+            ov::op::v0::Constant::create(ov::element::i32, {reshape_target_shape.size()}, reshape_target_shape);
+        last_node = std::make_shared<ov::op::v1::Reshape>(last_node, target_shape_node, false);
+    }
+    return last_node;
+}
+void GatherWeightsDecompression::generate_inputs(const std::vector<ov::Shape>& target_input_static_shapes) {
+    inputs.clear();
+    const auto& model_inputs = function->inputs();
+    for (size_t i = 0; i < model_inputs.size(); ++i) {
+        const auto& model_input = model_inputs[i];
+        ov::test::utils::InputGenerateData in_data;
+        in_data.start_from = -1;
+        in_data.range = 2;
+        in_data.resolution = 10000;
+        ov::Tensor tensor = ov::test::utils::create_and_fill_tensor(model_input.get_element_type(),
+                                                                    target_input_static_shapes[i],
+                                                                    in_data);
+        inputs.insert({model_input.get_node_shared_ptr(), tensor});
+    }
+}
+void GatherWeightsDecompression::check_results() {
+    const auto& test_param = GetParam();
+    ov::element::Type weights_precision = std::get<2>(test_param);
+    bool found_gather_compressed = false;
+    for (const auto& n : compiledModel.get_runtime_model()->get_ordered_ops()) {
+        if (n->get_friendly_name() == "Compressed_weights") {
+            ASSERT_EQ(n->get_output_element_type(0), weights_precision);
+        }
+        if (n->get_friendly_name() == "gather_node") {
+            // Input 4/5 means pattern matched.
+            if (n->get_input_size() == 4u || n->get_input_size() == 5u) {
+                found_gather_compressed = true;
+            }
+        }
+    }
+    EXPECT_TRUE(found_gather_compressed);
```

Comment:
```suggestion
    if (targetDevice != "GPU")
        EXPECT_TRUE(found_gather_compressed);
```
Because GPU plugin's `get_runtime_model()` does not have constant nodes, it's not possible to check the number of inputs for the `gather_node`. I think this check should be disabled for GPU.

Reply:
Don't agree. We should avoid device specific conditions in common tests.
Can we change validation condition? For example can we check that the op has type CompressedGather? I think it should be true for both plugins